src/Microsoft.AspNetCore.Sockets.Http/Internal/Transports/LongPollingTransport.cs

```diff
@@ -26,15 +26,67 @@ public LongPollingTransport(CancellationToken timeoutToken, PipeReader applicati
             _logger = loggerFactory.CreateLogger<LongPollingTransport>();
         }
 
+        public async Task ProcessFirstRequestAsync(HttpContext context, CancellationToken token)
+        {
+            try
+            {
+                await context.Response.Body.FlushAsync();
+
+                ReadResult result;
+                try
+                {
+                    result = await _application.ReadAsync(token);
+                }
+                catch (Exception)
+                {
+                    // We can't let the exception escape because we've already written headers, so
+                    // ASP.NET Core will just terminate the HTTP connection, which is bad.
+                    // The next poll will trigger the catch block below and throw.
+                }
+
+                var buffer = result.Buffer;
+
+                Log.LongPollingWritingMessage(_logger, buffer.Length);
+                try
+                {
+                    await context.Response.Body.WriteAsync(buffer);
+                }
+                finally
+                {
+                    _application.AdvanceTo(buffer.End);
+                }
+            }
+            catch (OperationCanceledException)
+            {
+                if (context.RequestAborted.IsCancellationRequested)
+                {
+                    // Don't count this as cancellation, this is normal as the poll can end due to the browser closing.
+                    // The background thread will eventually dispose this connection if it's inactive
+                    Log.LongPollingDisconnected(_logger);
+                }
+            }
+            catch (Exception ex)
+            {
+                Log.LongPollingTerminated(_logger, ex);
+            }
+        }
+
         public async Task ProcessRequestAsync(HttpContext context, CancellationToken token)
         {
             try
             {
                 var result = await _application.ReadAsync(token);
                 var buffer = result.Buffer;
 
+                // IF the buffer is empty and the read result is completed on the first request
+                // we still have to send a 200 because the headers have already been flushed.
                 if (buffer.IsEmpty && result.IsCompleted)
                 {
+                    // Complete the ReadAsync by indicating we haven't processed any data yet
+                    // (of course, there is no data because the buffer is empty, but we have to 
+                    // call AdvanceTo after every ReadAsync even if we don't have any data)
+                    //_application.AdvanceTo(buffer.Start);
```

Comment:
Why is this commented out?

Reply:
We needed this before when we were handling the first request in this method

Reply:
Actually, I think we still need this..

Reply:
Why?

Reply:
Technically we don't need it because in this branch when the 204 is returned the connection would get disposed. https://github.com/aspnet/SignalR/blob/dev/src/Microsoft.AspNetCore.Sockets.Http/HttpConnectionDispatcher.cs#L250-L253
I was correct before when I said that we needed it when we handled the first request in this method. If we didn't call advance an exception would be thrown. Unnecessary now